firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -197,25 +170,22 @@ public static FirebaseApp getInstance() {
    */
   @NonNull
   public static FirebaseApp getInstance(@NonNull String name) {
-    synchronized (LOCK) {
-      FirebaseApp firebaseApp = INSTANCES.get(normalize(name));
-      if (firebaseApp != null) {
-        return firebaseApp;
-      }
+    ComponentRuntime instance = ContainerOwner.getInstance(name);
 
-      List<String> availableAppNames = getAllAppNames();
-      String availableAppNamesMessage;
-      if (availableAppNames.isEmpty()) {
-        availableAppNamesMessage = "";
-      } else {
-        availableAppNamesMessage =
-            "Available app names: " + TextUtils.join(", ", availableAppNames);
-      }
-      String errorMessage =
-          String.format(
-              "FirebaseApp with name %s doesn't exist. %s", name, availableAppNamesMessage);
-      throw new IllegalStateException(errorMessage);
+    if (instance != null && instance.get(FirebaseApp.class) != null) {
+      return instance.get(FirebaseApp.class);
```

Comment:
Do we expect this to happen? getApps() does not check for existence of a FirebaseApp when adding it to the list.